subsys/audio_module/audio_module.h

```diff
@@ -0,0 +1,463 @@
+ */
+int audio_module_open(struct audio_module_parameters *parameters,
+		      struct audio_module_configuration *configuration, char *name,
+		      struct audio_module_context *context, struct audio_module_handle *handle);
+
+/**
+ * @brief Function to close an open module.
+ *
+ * @param handle[in/out]  The handle to the module instance
+ *
+ * @return 0 if successful, error otherwise
+ */
+int audio_module_close(struct audio_module_handle *handle);
+
+/**
+ * @brief Function to reconfigure a module.
+ *
+ * @param handle[in/out]     The handle to the module instance
+ * @param configuration[in]  Pointer to the module's configuration to set
+ *
+ * @return 0 if successful, error otherwise
+ */
+int audio_module_reconfigure(struct audio_module_handle *handle,
+			     struct audio_module_configuration *configuration);
+
+/**
+ * @brief Function to get the configuration of a module.
+ *
+ * @param handle[in/out]      The handle to the module instance
+ * @param configuration[out]  Pointer to the module's current configuration
+ *
+ * @return 0 if successful, error otherwise
+ */
+int audio_module_configuration_get(struct audio_module_handle *handle,
+				   struct audio_module_configuration *configuration);
+
+/**
+ * @brief Function to connect two modules together.
+ *
+ * @param handle_from[in/out]  The handle for the module for output
+ * @param handle_to[in]        The handle of the module for input. If it is the same as handle_from
+ *                             the data will be put on the handle_from->output_message_quere.
+ *
+ * @return 0 if successful, error otherwise
+ */
+int audio_module_connect(struct audio_module_handle *handle_from,
+			 struct audio_module_handle *handle_to);
+
+/**
+ * @brief Function to disconnect modules from each other.
+ *
+ * @param handle[in/out]         The handle for the module
+ * @param handle_disconnect[in]  The handle of the module to disconnect
+ *
+ * @return 0 if successful, error otherwise
+ */
+int audio_module_disconnect(struct audio_module_handle *handle,
+			    struct audio_module_handle *handle_disconnect);
+
+/**
+ * @brief Start processing data in the module given by handle.
+ *
+ * @param handle[in/out]  The handle for the module to start
+ *
+ * @return 0 if successful, error otherwise
+ */
+int audio_module_start(struct audio_module_handle *handle);
+
+/**
+ * @brief Stop processing data in the module given by handle.
+ *
+ * @param handle[in/out]  The handle for the module to be stopped
+ *
+ * @return 0 if successful, error otherwise
+ */
+int audio_module_stop(struct audio_module_handle *handle);
+
+/**
+ * @brief Send a data buffer to a module, all data is consumed by the module.
+ *
+ * @param handle[in/out]   The handle for the receiving module instance
+ * @param block[in]        Pointer to the audio data block to send to the module
+ * @param response_cb[in]  Pointer to a callback to run when the buffer is
+ *                         fully comsumed in a module
```

Comment:
```suggestion
 *                         fully consumed in a module
``` we will add a spell checker :)